typescript/cli/src/submit/submit.ts

```diff
@@ -40,32 +38,37 @@ export async function getSubmitterBuilder<TProtocol extends ProtocolType>({
 async function getSubmitter<TProtocol extends ProtocolType>(
   multiProvider: MultiProvider,
   submitterMetadata: SubmitterMetadata,
+  isDryRun = false,
 ): Promise<TxSubmitterInterface<TProtocol>> {
   switch (submitterMetadata.type) {
     case TxSubmitterType.JSON_RPC:
       return new EV5JsonRpcTxSubmitter(multiProvider);
     case TxSubmitterType.IMPERSONATED_ACCOUNT:
+      if (!isDryRun)
+        throw new Error(
+          'Impersonated account submitters may only be used during dry-runs.',
+        );
```

Comment:
I know you mentioned this but can you explain why this constraint is necessary?

Reply:
It's not completely necessary, but avoids moving implementation logic up into context creation. with that said, i did attempt this change and it's not too bad. let me know what you think

Reply:
also– we are now using the chain from the strategy to set the dry-run network :)